File: app/src/main/java/com/example/dzj/myreader/utils/ExecutorsUtil.java
package com.example.dzj.myreader.utils;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * 单例线程池
 */
public class ExecutorsUtil {

    private static volatile ExecutorsUtil executorsUtil;
    private ExecutorService fixedThreadPool;

    private ExecutorsUtil(){
        fixedThreadPool = Executors.newFixedThreadPool(5);
    }

    public static ExecutorsUtil getInstance(){
        ExecutorsUtil ex = executorsUtil;
        if(ex == null){
            synchronized (ExecutorsUtil.class){
                ex = new ExecutorsUtil();
                executorsUtil = ex;
            }
        }
        return ex;
    }

    public void execute(Runnable runnable){
        if(fixedThreadPool != null){
            fixedThreadPool.execute(runnable);
        }
    }
}

File: app/src/main/java/com/example/dzj/myreader/utils/ExecutorsUtil.kt
package com.example.dzj.myreader.utils

import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

/**
 * @author: ${User}
 * @date: ${Date}
 *
 */
class ExecutorsUtil private constructor() {

    private var fixedThreadPool: ExecutorService

    init {
        fixedThreadPool = Executors.newFixedThreadPool(4)
    }

    companion object {
        @Volatile
        internal var instance: ExecutorsUtil? = null

        fun getInstance() : ExecutorsUtil? {
            instance ?: synchronized(this) {
                instance ?: ExecutorsUtil().also { instance = it }
            }
            return instance
        }
    }

    fun execute(runnable: Runnable) {
        fixedThreadPool.execute(runnable)
    }
}